InvenTree/templates/js/translated/table_filters.js

```diff
@@ -355,6 +368,10 @@ function getAvailableTableFilters(tableKey) {
                 type: 'bool',
                 title: '{% trans "Assigned to me" %}',
             },
+            assigned_to: {
+                title: '{% trans "Responsible" %}',
+                options: ownersList,
```

Comment:
This is a nice approach, but the API request is going to be made every time the table is rendered (even when not necessary).

I'd say that we want to be able to dynamically fill out the "options" attribute for a table filter when the filter is added. This way that API request only fires when the user selects "owner" from the drop-down menu of available filters. Will need to extend the functionality of `table_filters.js` to achieve this.

Reply:
well, I thought about that, too... We could change the value for this selection option into a function callback in `table_filter.js --> getAvailableTableFilters()`, like so:

```js
[...]
           assigned_to: {
              title: '{% trans "Responsible" %}',
              options: function () { 
                  var ownersList = {};
                  inventreeGet(`/api/user/owner/`, {}, {
                      async: false,
                      success: function(response) {
                          for (key in response) {
                              var owner = response[key];
                              ownersList[owner.pk] = {
                                  key: owner.pk,
                                  value: `${owner.name} (${owner.label})`,
                              };
                          }
                      }
                  });
                  return ownersList;
              },
          },
[...]
```
and then extend the html generation in `filter.js --> setupFilterList()` to run that callback like this:
```js
        if (options instanceof Function){
            // run callback function for filter options
            options = options();
        }
```
Now we can use either a `list` as before, or a `function()` for the filter options.
I tried that and it works flawlessly - should I commit this?

Reply:
I added the proposed solution as a commit for better overview as https://github.com/inventree/InvenTree/pull/4408/commits/acc31a20e1fed4dba11726a91760ab505047fb3e - let me know if you like it or not, I'm open for alternatives as well...